go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -333,10 +333,16 @@ func (oc *Controller) deleteEgressFirewall(egressFirewallObj *egressfirewallapi.
 	deleteDNS := false
 	obj, loaded := oc.egressFirewalls.LoadAndDelete(egressFirewallObj.Namespace)
 	if !loaded {
-		return fmt.Errorf("there is no egressFirewall found in namespace %s", egressFirewallObj.Namespace)
+		return fmt.Errorf("there is no egressFirewall found in namespace %s",
+			egressFirewallObj.Namespace)
 	}
 
-	ef, _ := obj.(egressFirewall)
+	ef, ok := obj.(*egressFirewall)
+	if !ok {
+		return fmt.Errorf("type assertion to *egressFirewall failed for "+
+			"EgressFirewall %s of type %T in namespace %s.",
+			egressFirewallObj.Name, egressFirewallObj, egressFirewallObj.Namespace)
```

Comment:
@ricky-rav I think instead of `egressFirewallObj` here we want just `obj` right, since that's what we're trying to assert just above?

Reply:
That's true, thanks for spotting that! I just pushed a commit that fixes that.